zlib.patch

```diff
@@ -0,0 +1,15 @@
+Use a system includes instead of a local includes for zlib, as the 
+Bazel Central Registry version of zlib enforces that only system 
+includes are used.
+
+--- a/libs/iostreams/src/zlib.cpp
++++ b/libs/iostreams/src/zlib.cpp
+@@ -17,7 +17,7 @@
+ #include <boost/throw_exception.hpp>
+ #include <boost/iostreams/detail/config/dyn_link.hpp>
+ #include <boost/iostreams/filter/zlib.hpp> 
+-#include "zlib.h"   // Jean-loup Gailly's and Mark Adler's "zlib.h" header.
```

Comment:
I do not understand why this is necessary. `#include "zlib.h"` should work from a Bazel perspective. For instance take. a look here: `https://github.com/Vertexwahn/FlatlandRT/blob/main/devertexwahn/imaging/tests/third_party/zlib_test.cpp` This test passes since a few month successful.

Reply:
Maybe this is necessary because we want that zlib "leaks" to Boost without letting the layering check take notice of this...

Reply:
Yeah I think it's the layering check that catches this